onadata/apps/api/viewsets/user_profile_viewset.py

```diff
@@ -200,3 +206,66 @@ def monthly_submissions(self, request, *args, **kwargs):
 
         serializer = MonthlySubmissionsSerializer(instance_count, many=True)
         return Response(serializer.data[0])
+
+    @action(detail=False)
+    def verify_email(self, request, *args, **kwargs):
+        webhook = settings.EMAIL_VERIFICATION_WEBHOOK
```

Comment:
I get your concern and I had considered it while working on this feature. I need to add documentation for the this endpoints and the settings variables that need to be set. So if you checkout the if-block that follows [this line](https://github.com/onaio/onadata/pull/1442/files#diff-186b711876dec6613b24f328e6e02ea4R161), you'll notice that if `ENABLE_EMAIL_VERIFICATION` is not set to `true`, the email verification endpoints won't be accessible and a user will get a 403. This means current installations won't be affected. If you set it to `true`, you have to set all the other email verification settings as well for the feature to work. In my opinion, if the app fails with a 500, the dev team would know that they need to set the settings variables that are required. Returning a `None` if the variable does not exist won't be helpful because all those variables are required.

Reply:
I agree you should add API documentation for this endpoint.

However, on the 500 error, I would rather we define default behaviour if those values are not set and provide examples of how that can be defined as you have done on the docker settings file.

Reply:
Cool, I'll define a default behaviour.